src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -333,7 +333,9 @@ private static IEnumerable<MethodBodyStatement> BuildDeserializeBody(Parameter e
                     new SwitchStatement(discriminatorElement.GetString(), GetDiscriminatorCases(jsonElement, discriminator).ToArray())
                 };
             }
-            if (discriminator is not null && !serialization.Type.HasParent && discriminator.DefaultObjectType != null && !serialization.Type.Equals(discriminator.DefaultObjectType.Type))
+            // we redirect the deserialization to the `DefaultObjectType` (the unknown version of the discriminated set) if possible.
+            // We could only do this when there is a discriminator, and the discriminator does not have a value (having a value indicating it is the child instead of base), and there is an unknown default object type to fall back, and I am not that fallback type.
+            if (discriminator is { Value: null, DefaultObjectType: { } defaultObjectType } && !serialization.Type.Equals(defaultObjectType.Type))
```

Comment:
Our previous condition has a check `!serializationType.HasParent`, meaning that we do not use the unknown to deserialize when this type has a parent.

This is true for the normal models, and the derived types in a discriminated set. But for the base type in the discriminated set, it can have parent model.
The issue occurs here when the base class in the set has a base (without a discriminator), and it is abstract - because the condition here does not match and we will deserialize it as if it is a normal model and call its ctor which leads to the exception.